gui.py

```diff
@@ -12,99 +14,169 @@ def __init__(self, parent=None):
 
-        main_layout.addWidget(self.status_info_label)
-        self.update_game_gui(self.game_dir)
+        self.main_layout.addWidget(self.status_info_label)
         self.craft_list_ql.currentRowChanged.connect(self.updateList)
-        main_layout.addWidget(self.upload_btn)
-        main_layout.addWidget(self.close_btn)
-        self.setLayout(main_layout)
+        self.main_layout.addWidget(self.upload_btn)
+        self.setLayout(self.main_layout)
+
+        # Build manager and set ui
+        self.manager = KSP2Skfb(self.game_dir, True)
+        self.update_game_ui(self.game_dir)
+        self.enable_skfb_ui(False)
+        QtCore.QObject.connect(self.manager.emitter, QtCore.SIGNAL('building(QString, int, int)'), self.update_upload_btn)
+
+        # If bad path at launch, pop up the filedialog to locate it
+        if not os.path.exists(self.game_dir):
+            QtGui.QMessageBox.warning(self, "Game directory not found", "Please set your game directory")
+            self.search_game_directory()
+
+    def get_user_data(self):
+        if os.path.exists('userdata.json'):
+            need_clear = False
+            with open('userdata.json', 'r') as f:
+                try:
+                    self.user_data = json.load(f)
+                    print(self.user_data)
+                except:
+                    need_clear = True
+                    print(os.getcwd())
+            if need_clear:
+                os.remove('userdata.json')
+
+    def dump_user_data(self):
+        with open('userdata.json', 'w') as f:
+            json.dump(self.user_data, f)
+
+    def update_upload_btn(self, text, val, total):
+        if total > 0:
+            percent = int(float(val)/total*100)
+            self.upload_btn.setText("{} {}%".format(text, percent))
+        else:
+            self.upload_btn.setText("{}".format(text))
 
+        self.repaint()
+
+    def set_upload_btn_state(self, state=None):
+        ''' Update the upload btn status '''
+        if state == 'upload':
+            self.upload_btn.setText('Uploading...')
+            self.upload_btn.setStyleSheet("background-color: rgb(28, 170, 217); color: white;")
+        elif state == 'building':
+            self.upload_btn.setEnabled(True)
+            self.upload_btn.setText('Building zip archive...')
+            self.upload_btn.setStyleSheet("background-color: rgb(170, 170, 170); color: white;")
+        elif state =='publish':
+            self.upload_btn.setEnabled(True)
+            self.upload_btn.setText('Publish to Sketchfab')
+            self.upload_btn.setStyleSheet("background-color: rgb(28, 170, 217); color: white;")
+
+        self.repaint()
+
+    def description_input_limit(self):
+        if len(self.description_tb.toPlainText()) > 1024:
+            self.description_tb.setText(self.description_tb.toPlainText()[:1024])
+        self.description_label.setText('Description ({}/1024)'.format(len(self.description_tb.toPlainText())))
```

Comment:
not sure the `({}/1024)` is totally understandable from a user point of view. Maybe say 1024 characters?